pkg/controller/elasticsearch/version/zen2/voting_exclusions.go

```diff
@@ -19,6 +21,40 @@ var (
 	log = logf.Log.WithName("zen2")
 )
 
+const (
+	// VotingConfigExclusionsAnnotationName is an annotation that stores the last applied voting config exclusions.
+	// An empty value means no voting config exclusions are set.
+	VotingConfigExclusionsAnnotationName = "elasticsearch.k8s.elastic.co/voting-config-exclusions"
+)
+
+// serializeExcludedNodesForAnnotation returns a sorted comma-separated representation of the given slice.
+func serializeExcludedNodesForAnnotation(excludedNodes []string) string {
+	// sort a copy to not mutate the given slice
+	sliceCopy := make([]string, len(excludedNodes))
+	copy(sliceCopy, excludedNodes)
+	sort.Strings(sliceCopy)
+	return strings.Join(sliceCopy, ",")
+}
```

Comment:
I decided to sort a copy of the slice instead of mutating the slice itself. That's an extra allocation but I would not expect such a side effect to occur in that function.